### PR TITLE
Updated custom scripts in packages to match cross-platform behaviour

### PR DESCRIPTION
**Pull request by michaelnoonan**
See https://github.com/OctopusDeploy/Issues/issues/6267

I've tried to make this page clear, helpful, and not too long, nor duplicate too much information from other places. The page feels, on balance, to be a much more helpful single resource for doing this kind of work with Octopus.

### CODE REVIEW

File: docs/deployment-examples/custom-scripts/index.md [hunk 1]
```diff
@@ -11,8 +11,8 @@ As a convention-oriented deployment tool, Octopus can perform a number of action
 Octopus supports the following scripting environments:
 
  - PowerShell scripts (.ps1)
- - C# scripts (.csx) using [ScriptCS](https://github.com/scriptcs/scriptcs)
```
Comment (michaelnoonan) [inline docs/deployment-examples/custom-scripts/index.md]: A little hat-tip to Linux being more first-class. I wonder if we should change it any more, aligning it with the telemetry we see? Most popular at the top.

File: docs/deployment-examples/custom-scripts/scripts-in-packages/index.md [hunk 2]
```diff
@@ -19,21 +23,43 @@ Octopus will detect these scripts and invoke them at the appropriate time during
...
-
 ## Running a script when a step fails
 
 You can create a file named `DeployFailed.<ext>`, which will be invoked if the step fails. Our blog post about this feature [describes how DeployFailed.<ext> works](https://octopus.com/blog/deployfailed).
 
+## How Octopus executes your scripts
+
+At each stage during the deployment, Octopus will look for a scripts matching the current stage, and execute the first matching script it finds ordered by a platform-specific priority (see [cross-platform support](#cross-platform-support)).
+
+1. Octopus extracts the package to new uniquely named directory. **This becomes the current working directory.**
```
Comment (michaelnoonan) [inline docs/deployment-examples/custom-scripts/scripts-in-packages/index.md]: This is slightly duplicated with another page; but has the details pertinent to the script execution like the current working directory.

File: docs/deployment-examples/custom-scripts/scripts-in-packages/index.md [hunk 3]
```diff
@@ -19,21 +23,43 @@ Octopus will detect these scripts and invoke them at the appropriate time during
...
+### Cross-platform support {#cross-platform-support}
+
+If you are deploying the same package to multiple platforms, you can:
+
+1. Use a single scripting language common to all platforms. Octopus will run the single script using the same scripting runtime on all platforms.
+2. Use the scripting language most native to each platform. Octopus will run the most appropriate script for each platform using a platform-specific priority order.
+
+The platform-specific priority order Octopus uses to select scripts is:
+
+- Linux: Bash, PowerShell, C#, F#, Python
```
Comment (michaelnoonan) [inline docs/deployment-examples/custom-scripts/scripts-in-packages/index.md]: Hat-tip to Linux first.

Comment (michaelnoonan) [reply docs/deployment-examples/custom-scripts/scripts-in-packages/index.md]: Should we move Python further up the priority list?

File: docs/deployment-examples/custom-scripts/index.md (hunk 1 above)
Comment (wordlee) [reply docs/deployment-examples/custom-scripts/index.md]: That makes sense to me; it would mean making things easier for the most number of users.

**Comment by wordlee** [comment]
BTW, the build errors don't appear to be related to the content here. I'm following up.

**Pull request merged**